### PR TITLE
Add `live_isolated` to connect a LiveView in isolation

### PR DESCRIPTION
**Pull request by alexgaribay**
Allows a LiveView to be mounted and connected in isolation without needing an explicit route and page.

### CODE REVIEW

File: lib/phoenix_live_view/test/live_view_test.ex [hunk 1]
```diff
@@ -268,6 +271,37 @@ defmodule Phoenix.LiveViewTest do
...
+  Spawns a connected LiveView process mounted in isolation as the sole rendered element.
+
+  For LiveViews that are intended to be reused in multiple places, you can
+  connect a LiveView in isolation without needing a built a page that's tied to
+  a route. Testing pages that contains the LiveView to a degree is still
+  recommened.
+
+  ## Options
+
+    * `:connect_params` - the map of params available in connected mount mount.
```
Comment (gregjopa) [inline lib/phoenix_live_view/test/live_view_test.ex]:
```suggestion
    * `:connect_params` - the map of params available in connected mount.
```

Comment (alexgaribay) [reply lib/phoenix_live_view/test/live_view_test.ex]: Nice catch 👍

**Comment by benwilson512** [comment]
Can you elaborate on how this is different from live_render? Your example doesn't show where you'd call this macro.

**Comment by alexgaribay** [comment]
This macro is specifically for testing. Currently, a LiveView has to be reachable from routing to a specific page to be testable. For widget-style LiveViews, it awkward to test a page it's on without being able to test the widget itself. This macro allows you to render a live component in test without needing the LiveView to be attached to a specific route.

**Comment by benwilson512** [comment]
Ah, I missed what the module that it's in. This seems like it could be accomplished as a function right?

**Comment by alexgaribay** [comment]
It could if I added another argument for the endpoint. `Phoenix.ConnTest.get` needs the `@endpoint` attribute present for it to work. Since it's a test helper, I feel a macro is appropriate.

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**